From 65adc2305efb6a0590c3aac3332d96fcaaf906fa Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:32 -0400
Subject: [powerpc] Remove the unneeded trigger of decrementer interrupt in decrementer_check_overflow

Message-id: <1410545655-205645-104-git-send-email-dzickus@redhat.com>
Patchwork-id: 94268
O-Subject: [RHEL7 PATCH 103/626] powerpc: Remove the unneeded trigger of decrementer interrupt in decrementer_check_overflow
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 3139b0a797d6826519ed98a13623a92f12269613
Author: Kevin Hao <haokexin@gmail.com>
Date:   Wed Apr 17 17:50:35 2013 +0800

    powerpc: Remove the unneeded trigger of decrementer interrupt in decrementer_check_overflow

    Previously in order to handle the edge sensitive decrementers,
    we choose to set the decrementer to 1 to trigger a decrementer
    interrupt when re-enabling interrupts. But with the rework of the
    lazy EE, we would replay the decrementer interrupt when re-enabling
    interrupts if a decrementer interrupt occurs with irq soft-disabled.
    So there is no need to trigger a decrementer interrupt in this case
    any more.

    Signed-off-by: Kevin Hao <haokexin@gmail.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/irq.c b/arch/powerpc/kernel/irq.c
index 48c4e30..c7cb8c2 100644
--- a/arch/powerpc/kernel/irq.c
+++ b/arch/powerpc/kernel/irq.c
@@ -116,8 +116,6 @@ static inline notrace int decrementer_check_overflow(void)
   u64 now = get_tb_or_rtc();
   u64 *next_tb = &__get_cpu_var(decrementers_next_tb);
  
- if (now >= *next_tb)
-  set_dec(1);
  return now >= *next_tb;
 }
 
-- 
1.7.1